split-the-stream/README.md

```diff
@@ -0,0 +1,184 @@
+# Ideas about splitting the source-git work and upstream work
+
+Some ideas, possibilities, pros and cons of moving out the source-git related work.
+
+## What does the source-git workflow mean?
+
+Must have:
+
+- If user creates a merge-request on the source-git repository:
+  - Create a matching merge-request to the dist-git repository.
+  - Sync the CI results from the dist-git merge-request to the source-git merge-request.
+- If the dist-git is updated, update the source-git repository by opening a PR.
+- User is able to convert source-git change to the dist-git change locally via CLI.
+
+Should have:
+
+- If the source-git merge-request is updated, update the dist-git merge-request.
+- If the source-git merge-request is closed, close the dist-git merge-request.
+
+Could have:
+
+- User is able to re-trigger the dist-git CI from the source-git merge-request.
+- User is able to re-create the dist-git CI from the source-git merge-request.
+
+## Key questions:
+
+1. Should we start developing a new service or modify the existing packit-service to be able to deploy only with a gitlab-endpoint?
+2. How to link merge requests in the src namespace to the ones in the rpms namespace using the GitLab API? This should be bidirectional.
+3. Check the GitLab API to learn more about working with merge-trains and pipelines, in order to support the UX for merging source-git MRs (see the doc linked bellow).
+4. How are CI results going to be displayed in dist-git MRs? We wan't to know this so that we can think about ways to take those results and display them for contributors on the source-git MRs.
+
+## Split
+
+We have multiple options:
+
+### 0. no split
+
+- No extra cost of two deployments and two codebases.
+- New jobs will be implemented as new handlers.
+- We don't have support for multiple identities for one forge.
+  - Not hard to do but requires some work.
+- Fedora-source-git friendly: easy combination of events (different for Fedora and Stream)
+  and handlers (=implementation, can be shared).
+
+### 1. same codebase, new deployment
+
+- No extra cost of maintenance of two codebases.
+- New jobs are implemented as new handlers.
+- Different identities can be used in one git forge (=gitlab.com).
+- Resources can be tweaked separately.
+- Fedora-source-git friendly: easy combination of events (different for Fedora and Stream)
+  and handlers (=implementation, can be shared).
+
+### 2. separate workers
```

Comment:
this look really intriguing to be honest - those workers could then have a lifecycle on their own